WeWatch/WeWatch/Custom TabBar/TabBarViewModel.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  TabBarViewModel.swift
+//  WeWatch
+//
+//  Created by Anton on 30/12/2024.
+//
+
+import Foundation
+import SwiftUI
+
+internal final class TabBarViewModel: ObservableObject {
+    
+    @Published internal var homeViewIsActive: Bool = true
+    @Published internal var searchViewIsActive: Bool = false
+    @Published internal var bookmarkViewIsActive: Bool = false
+    @Published internal var selectedTab: TabViewType = .homeView
+    
+    internal enum TabViewType {
+        
+        case bookmark
+        case searchView
+        case homeView
+        
+        internal func activeView() -> AnyView {
+            switch self {
+            case.bookmark:
+                return AnyView(BookmarkView())
+            case.searchView:
+                return AnyView(SearchView())
+            case.homeView:
+                return AnyView(HomeView())
+            }
+        }
```

Comment:
You can remove it as well. We can approach it slightly differently while still preserving the main purpose of this enum.